src/users.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use anyhow::Error;
+use bitcoin::consensus::Encodable;
+use hyper::{header::HeaderValue, StatusCode};
+use serde_json::Value;
+
+use crate::client::{
+    GenericRpcMethod, RpcError, RpcMethod, RpcRequest, RpcResponse, METHOD_NOT_ALLOWED_ERROR_CODE,
+    METHOD_NOT_ALLOWED_ERROR_MESSAGE, MISC_ERROR_CODE, PRUNE_ERROR_MESSAGE,
+};
+use crate::env::Env;
+use crate::fetch_blocks::fetch_block;
+use crate::rpc_methods::{
+    GetBlock, GetBlockHeader, GetBlockHeaderParams, GetBlockResult, GetBlockchainInfo,
+};
+
+#[cfg(feature = "compat")]
+use crate::util::compat::StrCompat;
+
+#[derive(Debug, serde::Deserialize)]
+pub struct Users(pub HashMap<String, User>);
+impl Users {
+    pub fn get(&self, auth: &HeaderValue) -> Option<(String, &User)> {
+        let header_str = auth.to_str().ok()?;
+        let auth = header_str.strip_prefix("Basic ")?;
+        let auth_decoded = base64::decode(auth).ok()?;
+        let auth_decoded_str = std::str::from_utf8(&auth_decoded).ok()?;
+        let mut auth_split = auth_decoded_str.split(":");
+        let name = auth_split.next()?;
+        let pass = auth_split.next()?;
+        self.0
+            .get(name)
+            .filter(|u| u.password == pass)
+            .map(|u| (name.to_owned(), u))
+    }
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
```

Comment:
Renaming this is backwards-incompatible and breaks expectations of `configure_me` behavior. Please don't rename it.

Reply:
yeah, sorry, did that because of yaml convention. It would be nice to have serde be able to let the backend choose the casing, such that json can get `camelCase`, yaml can get `kebab-case`, etc without changing the serialize impl. I can revert that here, and move the translation to my configuration code, no reason these kinds of opinions should be buried deep in the codebase like that.

Reply:
Yeah that makes sense. Over time it occurred to me that `serde` is more suitable at the boundaries and whatever it deserializes should be converted to internal representation. It may look weird but I don't know of a better approach.